### DISCUSSION TITLE
How to pass parameters to a dashboards meta data (title and output-file)?

### DISCUSSION THREAD

**Discussion by brianmcdonald** [quarto-dev/quarto-cli#10374] [Q&A] [answered]
### Description

I would like to produce multiple dashboards, each with custom text in the title and the output-file name, which are passed by the parameter option(or similar) when running quarto render eg `quarto render -P zone:"example_text"

While the Parameter option works well for parameters within the body of the analysis, it doesn't seem to work for the meta data. Similarly Variables, Meta and Env, allow for placing text in the metadata but not in a parameterized manner that allows multiple dashboards from multiple inputs.

Below is my document code. Any solutions or ideas are welcome.

````cmd
---
title: Zone name - {{< meta params.zone >}}
author: ''
subtitle: Zonal Profile
title-block-banner: '#1B657C'

output-file: {{< meta params.zone >}}
lightbox: true
format:
  dashboard:
    logo: input/logo.svg
    include-in-header:
      text: <link rel="icon" type="image/x-icon" href="favicon.ico">
    orientation: columns
    theme:
      - input/Cluster.scss
---

# Page 1

### Column {width="65%"}

#### Row {.tabset}

##### Tab 1

test1
```{python}
#| tags: [parameters]
zone = "testZone"
```
````

**Comment by brianmcdonald** [comment]
A discussion on this is [here](https://github.com/quarto-dev/quarto-cli/discussions/5654)

It seems that [Profiles](https://quarto.org/docs/projects/profiles.html) can achieve what I'm looking for.

**Reply by mcanouil** [reply, marked as answer]
There are multiple threads already discussing this extensively.

- #10081
- #10234

And many more.

**Reply by brianmcdonald** [reply]
Thanks @mcanouil that second link was very useful

What worked best for me in the end was to use the -M flag, -o flag and -P flags all together when rendering the project.
-M to set the title, -o to set the output filename, and -P for parameters  

I preferred passing them on the command line rather than generating many yml files

**Reply by mcanouil** [reply]
Glad you're found a way that suits you!

**Reply by mcanouil** [reply]
@hasibagen Please avoid posting on all related discussions, especially when the solution was already provided. This trigger a lot of notifications.

**Comment by waqi786** [comment]
To produce multiple dashboards with custom text in the title and the output file name using Quarto, you can leverage Quarto's parameterization feature to dynamically set metadata values. Although parameters work well within the body of the analysis, they can also be used to set metadata values by utilizing Quarto’s YAML syntax and a pre-render script. First, define your Quarto document with a placeholder for the zone parameter in the YAML metadata section. For example, your document might look like this: --- title: "Zone name - {{zone}}" author: '' subtitle: Zonal Profile title-block-banner: '#1B657C' output-file: "{{zone}}" lightbox: true format: dashboard: logo: input/logo.svg include-in-header: text: <link rel="icon" type="image/x-icon" href="favicon.ico"> orientation: columns theme: - input/Cluster.scss --- # Page 1 ### Column {width="65%"} #### Row {.tabset} ##### Tab 1 test1 ```{python} #| tags: [parameters] zone = "testZone" ``` .

Next, create a Quarto pre-render script to replace the placeholders in the YAML metadata with the parameter values. Save this script as pre-render.js in the same directory as your Quarto document: const fs = require('fs'); let fileContent = fs.readFileSync('your_document.qmd', 'utf8'); const zone = process.env.ZONE; fileContent = fileContent.replace(/{{zone}}/g, zone); fs.writeFileSync('your_document.qmd', fileContent, 'utf8'); . In your shell, set the ZONE environment variable and run the Quarto render command with the pre-render script: export ZONE="example_text" node pre-render.js quarto render your_document.qmd -P zone:"$ZONE" . This process ensures that the zone parameter is dynamically inserted into the metadata of your Quarto document before rendering. By using this method, you can generate multiple dashboards with custom titles and output file names based on different zone parameter values. Repeat this process with different ZONE values to produce different dashboards as needed.

**Reply by mcanouil** [reply]
When using AI to provide answer, please take at least the time to format the answer using code block to make it legible.

To the future readers, the above answer is not the Quarto way.

@waqi786 if you want to reply, please use your own knowledge and don't copy/paste AI answers. Thank you!
